editor/src/messages/portfolio/document/overlays/utility_types.rs

```diff
@@ -72,6 +72,36 @@ impl OverlayContext {
 		self.render_context.line_to(end.x, end.y);
 		self.render_context.set_stroke_style_str(color.unwrap_or(COLOR_OVERLAY_BLUE));
 		self.render_context.stroke();
+		self.render_context
+			.set_line_dash(&JsValue::from(js_sys::Array::new()))
+			.map_err(|error| log::warn!("Error drawing dashed line: {:?}", error))
+			.ok();
+	}
+
+	pub fn dashed_line_with_pattern(&mut self, start: DVec2, end: DVec2, color: Option<&str>, dash_width: f64, gap_width: f64) {
```

Comment:
What's your reasoning for making two similar functions? I think we could reduce the lines of code by just making the one function more customizable to encompass all the requirements.

Reply:
The pre-existing function seems to be incorrectly defined the `dash_width` is being used as the 2nd parameter of the array which effectively makes it the length of the `gap` between two dashes of the line. and the actual dash width (the 1st parameter) seems to be hard set at `1` . 
And this function seems to be used at multiple other places, so i decided to make an extra function that exposes both gap_width and dash_width as individual parameters. ( since a 2px dash was needed to be followed with  a 2px blank space ).

Reply:
I see, thanks for the explanation. Can you please refactor it to use one function, preserving the appearance of existing usages (where the width is `gap` and `1` functionality you described)?

Reply:
sure!